src/actions.rs

```diff
@@ -339,9 +339,11 @@ impl EGraph {
                     let variants = values[1].bits as i64;
                     if variants == 0 {
                         let (cost, term) = self.extract(values[0], &mut termdag, sort);
-                        let extracted = termdag.to_string(&term);
-                        log::info!("extracted with cost {cost}: {extracted}");
-                        self.print_msg(extracted);
+                        if self.messages_enabled() {
```

Comment:
Why do you have these guards? `print_msg` won't do anything if messages are disabled, and the `to_string` PR is in already, right?

Reply:
I think it depends on the current goal for adding this benchmark. 

If the goal is to benchmark extraction itself (which still takes 400 ms - not as pathologically bad as it used to be, but arguably still a lot for a problem of this side), it might be helpful to disable stratification. Converting dag to a string still dominates runtime - it takes around a second and generates 889mb long string.

If the goal is to verify that pathologically bad behaviour doesn't occur anymore, it's probably okay to re-enable message printing.

In any case, having the ability to disable message printing is quite valuable in itself.

Reply:
Yeah, this PR mixes two things:

1. Allowing the e-graph to be run without collecting any messages and omits turning a term dag into a string
2. Run some benchmarks with this mode

The first is useful for situations where you are embedding egglog, like in the Python bindings, and you can get the extraction report instead of relying on the string output to know what's been extracted.

For the second, yeah maybe it's not necessary now that the performance is more reasonable for stringifying. On the other hand, maybe it does make sense to run our all our Python created benchmarks under this mode, if we want to closer track that performance...

Reply:
I updated the PR to run all tests with messages enabled, since now it's not too slow, but also switched the benchmarks to run the Python files with them disabled, to better track that performance and the slowdowns relevant to the upstream Python code.